### ISSUE TITLE
pycharm remote debug

### ISSUE THREAD

**Issue by positiveczp** [20tab/UnrealEnginePython#527]
how to use pycharm remote debug in python console, my code is  

def test():
    import os
    import sys
    sys.path.append(os.path.join(os.path.dirname(__file__), 'pycharm-debug'))
    sys.path.append(os.path.join(os.path.dirname(__file__), 'pycharm-debug.egg'))
    print 'pycharm_debug init'
    import pydevd
    pydevd.settrace('127.0.0.1', port=5555, stdoutToServer=True, stderrToServer=True, suspend=False)
    print 'pycharm_debug ok'

test()

when running this code, the ue4 will block

**Comment by positiveczp** [comment]
why never try it ? 
how can i debug use pycharm?

**Comment by rdeioris** [comment]
"i" never tried it :) Well just try my suggestion, i do not use pycharm, any feedback will be useful

**Comment by positiveczp** [comment]
that is say : we can not use remote debug with pycharm

**Comment by rdeioris** [comment]
Sorry i do not understand what you mean. I gave you a possible solution, have you tried it ?

**Comment by positiveczp** [comment]
a possible solution ? try adding the breakpoint in a new python thread?

**Comment by rdeioris** [comment]
yep, it is a common pattern, if you want ue to continue running you cannot block it. Using a different python thread will allow you to spawn the debug server and eventually attach to the right thread from the debugger client. This works in plain pdb, no idea for pycharm

**Comment by pilgrim333** [comment]
Did you get pycharm remote debug to work with UE4?  If you did, can you share?  Thank you.
